### PR TITLE
Ft/swap/add input modal

### PR DESCRIPTION
**Pull request by Xzirez**

### CODE REVIEW

**Comment by Jennimar17** [comment]
image
image
image

File: src/components/modals/SwapCoins/InputModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+const InputModal = ({
+  avaliableCoins,
+  continueButtonOnClick = () => {},
+  onClickClose = () => {},
+  getSwapRate = () => {},
+}: InputModalProps) => {
+  const [swapFrom, setSwapFrom] = useState<Swap>();
+  const [swapTo, setSwapTo] = useState<Swap>();
+  const [intervalId, setIntervalId] = useState<any>();
+  const rate = getSwapRate({ swapFrom: swapFrom, swapTo: swapTo });
```
Comment (cwagner22) [inline src/components/modals/SwapCoins/InputModal/index.tsx:20]: Use useMemo

File: src/components/modals/SwapCoins/InputModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  const swapFromChanged = useCallback(
+    ({ token, value }: Swap) => {
+      setSwapFrom({ token: token, value: value });
+    },
+    [swapFrom],
+  );
+
+  const swapToChanged = useCallback(
+    ({ token }: Swap) => {
```
Comment (cwagner22) [inline src/components/modals/SwapCoins/InputModal/index.tsx:34]: Probably better to extract the setInterval in an useEffect: https://upmostly.com/tutorials/setinterval-in-react-components-using-hooks
But I don’t understand what you want to do with the rate

**Comment by Jennimar17** [comment]
image

**Comment by Jennimar17** [comment]
image

File: src/components/modals/SwapCoins/InputModal/index.tsx (hunk 1 above)
Comment (Xzirez) [reply src/components/modals/SwapCoins/InputModal/index.tsx:20]: Ah this is when it should be useMemo. Gotcha.

**Pull request merged**